test/Semantics/resolve72.f90

```diff
@@ -19,7 +19,7 @@ subroutine inner1(arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8)
         type(*), pointer :: arg6
         !ERROR: Assumed-type argument 'arg7' cannot have the VALUE attribute
         type(*), value :: arg7
-        !ERROR: Assumed-type argument 'arg8' must be assumed shape or assumed size array
+        !ERROR: Assumed-type argument 'arg8' may not be an explicit-shape array
```

Comment:
I'm not sure that the new message is an improvement.  The existing message implies how the problem can be fixed.

Reply:
OK, I've changed it back and added the "assumed rank" part.